diff --git a/src/tool_operate.c b/src/tool_operate.c
index e6ca575..4f3e65f 100644
--- a/src/tool_operate.c
+++ b/src/tool_operate.c
@@ -622,6 +622,11 @@ static CURLcode post_per_transfer(struct GlobalConfig *global,
   free(per->outfile);
   free(per->uploadfile);

+  /* return exit code 1 when flag --fail is used */
+  if (config->failonerror && (CURLE_HTTP_RETURNED_ERROR == result)) {
+    result = CURLE_UNSUPPORTED_PROTOCOL;
+  }
+
   return result;
 }

