diff --git a/src/tool_operate.c b/src/tool_operate.c
index a1212c9..16334d3 100644
--- a/src/tool_operate.c
+++ b/src/tool_operate.c
@@ -772,6 +772,12 @@ skip:
     free(per->errorbuffer);
   curl_slist_free_all(per->hdrcbdata.headlist);
   per->hdrcbdata.headlist = NULL;
+
+  /* return exit code 1 when flag --fail is used */
+  if (config->failonerror && (CURLE_HTTP_RETURNED_ERROR == result)) {
+    result = CURLE_UNSUPPORTED_PROTOCOL;
+  }
+
   return result;
 }
 
