diff --git a/src/Makefile.am b/src/Makefile.am
index 7343731..52ff4cd 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -66,6 +66,10 @@ else
 curl_LDADD = $(top_builddir)/lib/libcurl.la @NSS_LIBS@ @SSL_LIBS@ @ZLIB_LIBS@ @CURL_NETWORK_AND_TIME_LIBS@
 endif

+# https://github.com/curl/curl/pull/7476
+curl_LDFLAGS = $(curl_LDFLAGS)
+curl_CPPFLAGS = $(AM_CPPFLAGS)
+
 # if unit tests are enabled, build a static library to link them with
 if BUILD_UNITTESTS
 noinst_LTLIBRARIES = libcurltool.la
